lib/std/fs/File.zig

```diff
@@ -1493,7 +1493,13 @@ pub fn lock(file: File, l: Lock) LockError!void {
     if (is_windows) {
         var io_status_block: windows.IO_STATUS_BLOCK = undefined;
         const exclusive = switch (l) {
-            .none => return,
+            .none => return windows.UnlockFile(
+                file.handle,
+                &io_status_block,
+                &range_off,
+                &range_len,
+                null,
+            ) catch error.Unexpected,
```

Comment:
I don't know about the validity of the rest of this PR but this error handling is inadequate.

Reply:
Would you prefer `unreachable` be used here in line with `unlock`?

Reply:
You should catch the error and then switch on it. And based on other uses of the function mark both possible errors as unreachable.

Reply:
Both errors are now unreachable, in line with `unlock`. There might be a case for returning `error.Unexpected` to the application if one doesn't consider `lock(Lock.none)` resource deallocation.

Reply:
The same applies to the other call.

Reply:
Of course. Thanks.